transactions/src/main/java/org/ehcache/transactions/xa/XAStore.java

```diff
@@ -838,7 +842,10 @@ public void releaseStore(Store<?, ?> resource) {
         // release the underlying store first, as it may still need the serializer to flush down to lower tiers
         underlyingStoreProvider.releaseStore(xaStore.underlyingStore);
         try {
-          helper.softLockSerializerRef.getAndSet(null).close();
+          Serializer<SoftLock> serializer = helper.softLockSerializerRef.getAndSet(null);
```

Comment:
Like I said, the type here can be restricted to `SoftLockSerializer` by changing the type of the AtomicReference and once done, you know it is not `Closeable` so the code below is useless.